### PR TITLE
Add map_index to RenderedTaskInstanceFields

### PR DESCRIPTION
**Pull request by ashb**
In order to do this "properly" we have also migrated it from using execution_date to run_id in its PK

### CODE REVIEW

**Comment by ashb** [comment]
@uranusjr I'm thinking we should add a `mapped_value` JSON column here in this migration. When does the mapped value become available?

File: airflow/migrations/versions/4eaab2fe6582_migrate_rtif_to_use_run_id_and_map_index.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    dag_run = sa.Table(
+        'dag_run',
+        metadata,
+        sa.Column('dag_id', StringID()),
+        sa.Column('run_id', StringID()),
+        sa.Column('execution_date', TIMESTAMP),
+    )
+
+
+def get_table_constraints(conn, table_name):
```
Comment (ephraimbuddy) [inline airflow/migrations/versions/4eaab2fe6582_migrate_rtif_to_use_run_id_and_map_index.py]: We now have this function in airflow/migrations/util.py as `get_mssql_table_constraints`

Comment (ashb) [reply airflow/migrations/versions/4eaab2fe6582_migrate_rtif_to_use_run_id_and_map_index.py]: Ah cool.

**Comment by uranusjr** [comment]
> I'm thinking we should add a mapped_value JSON column here in this migration. When does the mapped value become available?

They are populated when XCom is resolved, and should be available when RenderedTaskInstanceFields are written. However, since a RenderedTaskInstanceFields row maps to a TaskInstance row, I think we should simply resolve the mapped values directly instead.

On UI we should also add a marker to indicate a particular field value is mapped, so perhaps add a field to record the field _names_. Or maybe even this is unnecessary since this information is available on the task object? Or maybe in some scenarios we want to read RenderedTaskInstanceFields without parsing the DAG? (Currently I can only find this being used in `/rendered-k8s` and `/rendered-templates`, and in both views we already fetch the original task object and know what fields are mapped anyway.)

**Review by uranusjr — APPROVED** [review]
Assuming we don’t need to add mapping info to RenderedTaskInstanceFields (I don’t think we do), this should be good to go.

**Comment by ashb** [comment]
> Or maybe in some scenarios we want to read RenderedTaskInstanceFields without parsing the DAG?

The use case i am thinking of is for showing a table of all the mapped TIs in a detail view in the UI. And mapping can apply to things that aren't marked as templated

**Comment by ashb** [comment]
> Assuming we don’t need to add mapping info to RenderedTaskInstanceFields (I don’t think we do), this should be good to go.

Even if we do we can alter this migration in a future PR.

**Comment by uranusjr** [comment]
```
sqlalchemy.exc.OperationalError: (MySQLdb._exceptions.OperationalError) (1025, "Error on rename of './airflow/#sql-1_b' to './airflow/rendered_task_instance_fields' (errno: 150 - Foreign key constraint is incorrectly formed)")
```

MySQL is not happy…

**Comment by ashb** [comment]
Okay, I'm confused -- the mysql tests are not behaving like this for me locally

**Pull request merged**